src/components/CreateProvider.svelte

```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
+    import {dotApi, storeCurrentAction, transactionSigningAddress} from "$lib/stores";
+    import type {DotApi} from '$lib/storeTypes';
+    import {defaultDotApi} from "$lib/storeTypes";
+    import type {ApiPromise} from "@polkadot/api";
+    import {isLocalhost} from "$lib/utils";
+    import {submitCreateProvider} from "$lib/connections";
+    import TransactionStatus from "$components/TransactionStatus.svelte";
+    import {isFunction} from "@polkadot/util";
+    import {onMount} from "svelte";
+
+    let newProviderName = '';
+    let localDotApi: DotApi = defaultDotApi;
+    let web3FromSource;
+    let web3Enable;
+    let showTransactionStatus = false;
+    export let txnStatuses: Array<string> = [];
+    export let validAccounts = {}
+    export let signingAddress = '';
+    export let cancelAction;
+
+    onMount(async () => {
+        const extension = await import('@polkadot/extension-dapp');
+        web3FromSource = extension.web3FromSource;
+        web3Enable = extension.web3Enable;
+    });
+
+    let localSigningAddress = '0xabcdefbeef';
+    transactionSigningAddress.subscribe(addr => localSigningAddress = addr)
+    dotApi.subscribe((api: DotApi) => localDotApi = api)
+
+    const doCreateProvider = async (_evt: Event) => {
```

Comment:
Should this do something after running to update the page state that the provider exists now?

Reply:
yeah, probably.

Reply:
Now refreshes two stores which causes the top components to rerender.  Could be better but at least the display is correct.